src/test/claimtriebranching_tests.cpp

```diff
@@ -1053,15 +1060,1976 @@ BOOST_AUTO_TEST_CASE(claimtriebranching_hardfork_disktest)
     // increment to original expiration, should not be expired
     fixture.IncrementBlocks(fixture.originalExpiration - height_of_update_before_expiration);
     BOOST_CHECK(is_best_claim("test2", u2));
-    BOOST_CHECK(best_claim_effective_amount_equals("test2",2));
+    BOOST_CHECK(best_claim_effective_amount_equals("test2", 2));
     fixture.DecrementBlocks(fixture.originalExpiration - height_of_update_before_expiration);
     // increment to extended expiration, should be expired and not one block before
     fixture.IncrementBlocks(fixture.extendedExpiration - height_of_update_before_expiration);
     BOOST_CHECK(!is_best_claim("test2", u2));
     fixture.DecrementBlocks(fixture.extendedExpiration - height_of_update_before_expiration);
-    fixture.IncrementBlocks(fixture.extendedExpiration - height_of_update_before_expiration-1);
+    fixture.IncrementBlocks(fixture.extendedExpiration - height_of_update_before_expiration - 1);
     BOOST_CHECK(is_best_claim("test2", u2));
-    BOOST_CHECK(best_claim_effective_amount_equals("test2",1)); // the support expires one block before
+    BOOST_CHECK(best_claim_effective_amount_equals("test2", 1)); // the support expires one block before
+}
+
+BOOST_AUTO_TEST_CASE(insert_update_claim_test)
+{
+    ClaimTrieChainFixture fixture;
+
+    std::string sName1("atest");
+    std::string sName2("btest");
+    std::string sName3("atest123");
+    std::string sValue1("testa");
+    std::string sValue2("testb");
+    std::string sValue3("123testa");
+
+    uint256 hash0(uint256S("0000000000000000000000000000000000000000000000000000000000000001"));
+    BOOST_CHECK(pclaimTrie->getMerkleHash() == hash0);
+
+
+    CMutableTransaction tx1 = BuildTransaction(fixture.GetCoinbase());
+    tx1.vout[0].scriptPubKey = CScript() << OP_CLAIM_NAME
+                                         << std::vector<unsigned char>(sName1.begin(), sName1.end())
+                                         << std::vector<unsigned char>(sValue1.begin(), sValue2.end()) << OP_2DROP << OP_DROP << OP_TRUE;
```

Comment:
This mismatch on the end is what was causing the Travis-CI test failure. Strange thing: it doesn't fail with the GCC 7 builds, with or without optimizations. Only the GCC 4 builds fail.

Reply:
Interesting.  If you're able to track down why that is, it may save us some headaches in the future.  Once we are inline with upstream, gcc 5/6/7/8 becomes a viable option, perhaps even a minimum supported version.

Reply:
This looks like a bug? ```std::vector<unsigned char>(sValue1.begin(), sValue2.end())```

Reply:
Also, does the construction form make any difference?

Changing something like:
```
std::vector<unsigned char>(sName1.begin(), sName1.end())
```

to

```
std::vector<unsigned char>(sName1.data(), sName1.data() + sName1.size())
```

Reply:
On second read ... are you just saying that the mismatch is/what the actual problem and later compilers don't notice it?  We can play with the warning levels if so later.

Reply:
Ah right, it would require a temporary, (I hacked it since we're not using c++11).  The preferred syntax (c++11) would be:

```
{ sName1.data(), sName1.data() + sName1.size() }
```

The test without is something like:

```
std::vector<unsigned char> foo(sName1.data(), sName1.data() + sName1.size())

...
<< foo << other_stuff;
```

Reply:
BTW, if the test passes with that bug, there's another bug at the surface.

Reply:
I didn't notice the `data()` on the first parameter previously. This does compile and also does not throw an error in GCC7: `... std::vector<unsigned char>(sValue1.data(), sValue2.data() + sValue2.size()) ...`. It probably works because sName1 and sName2 are in order on the stack; you end up with sValue1 prepended to sValue2 with some junk in the middle, but we never verify the value.